utt/utt-client-api/include/utt-client-api/User.hpp

```diff
@@ -95,46 +103,64 @@ class User {
    */
   uint64_t getLastExecutedTxNum() const;
 
-  /**
-   * @brief Update the state of the client by applying a new transaction
-   *
-   * @param txNum The transaction number
-   * @param tx The transaction object
-   * @param sigs A vector of signatures for each output of the transaction
-   *
-   * @return true if the transaction was applied to the state
-   */
-  bool update(uint64_t txNum, const Tx& tx, const std::vector<std::vector<uint8_t>>& sigs);
-
-  /**
-   * @brief The client can ignore irrelevant transactions by recording the transaction number as a no-op.
-   *
-   */
-  void update(uint64_t txNum);
-
-  /**
-   * @brief Ask to burn some amount of tokens. This function needs to be called repeatedly until the final burn
-   * transaction is produced.
-   *
-   * @param amount The amount of tokens to burn.
-   *
-   * @return Returns a valid UTT transaction to burn or merge/split UTT tokens or an error.
-   *
-   */
+  /// @brief Update the user's state with the effects of a transfer transaction
+  /// @param txNum The transaction number
+  /// @param tx A transfer transaction
+  /// @param sigs The signatures on the transaction outputs
+  /// @return
+  bool updateTransferTx(uint64_t txNum, const utt::Transaction& tx, const utt::TxOutputSigs& sigs);
+
+  /// @brief Update the user's state with the effects of a mint transaction
+  /// @param txNum The transaction number
+  /// @param tx A mint transaction
+  /// @param sig The signature on the transaction output (we assume a mint tx has a single output)
+  /// @return
+  bool updateMintTx(uint64_t txNum, const utt::Transaction& tx, const utt::TxOutputSig& sig);
+
+  /// @brief Update the user's state with the effects of a burn transaction
+  /// @param txNum The transaction number
+  /// @param tx A burn transaction
+  /// @return
+  bool updateBurnTx(uint64_t txNum, const utt::Transaction& tx);
+
+  /// @brief The user records the tx as a no-op and skips it
+  /// @param txNum
+  bool updateNoOp(uint64_t txNum);
```

Comment:
Isn't it cleaner to return `void` and raise an exception if something goes wrong?

Reply:
Yes, I'm still undecided how to do the error handling for the client library and it's inconsistent at places. This will be addressed in future work.